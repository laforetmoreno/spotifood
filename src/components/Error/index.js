import React from 'react';
import { string } from 'prop-types';
import Button from 'components/Button';
import errorImage from '../../assets/error.svg';

import styles from './index.module.scss';

const Error = ({ message }) => (
  <div data-testid="error" className={styles.wrapper}>
    <span className={styles.message}>{message}</span>
    <img className={styles.image} alt={message} src={errorImage} />
    <Button onClick={() => window.location.reload()}>Tentar novamente</Button>
  </div>
);

Error.propTypes = {
  message: string,
};

Error.defaultProps = {
  message: 'Ops',
};

export default Error;
