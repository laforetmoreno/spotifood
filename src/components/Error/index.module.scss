@import '../../styles';

.wrapper {
  min-height: calc(100vh - 20px);
  @include flex-default();
  @include flex-column();
}

.image {
  max-width: rem(200px);
  margin-top: rem(100px);
  margin-bottom: rem(50px);
}

.message {
  font-size: rem(36px);
  color: $--color-green;
}
