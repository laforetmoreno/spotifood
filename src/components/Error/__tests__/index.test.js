import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Error from '../index';

describe('<Error />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Error />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(<Error />);
    const fieldNode = await waitForElement(() => getByTestId('error'));
    fireEvent.change(fieldNode);
  });
});
