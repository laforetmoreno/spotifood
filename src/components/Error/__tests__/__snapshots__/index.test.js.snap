// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<div
  className="wrapper"
  data-testid="error"
>
  <span
    className="message"
  >
    Ops
  </span>
  <img
    alt="Ops"
    className="image"
    src="error.svg"
  />
  <button
    className="wrapper"
    data-testid="btn"
    onClick={[Function]}
    type="button"
  >
    Tentar novamente
  </button>
</div>
`;
